### PR TITLE
Probability parameter in RandomHorizontalFlip, RandomHorizontalFlip

### PR DESCRIPTION
**Pull request by vfdev-5**
Addresses issue #414

### CODE REVIEW

File: torchvision/transforms/transforms.py [hunk 1]
```diff
@@ -331,31 +338,39 @@ def __call__(self, img):
...
+    """
+
+    def __init__(self, p=0.5):
+        self.p = p
 
     def __call__(self, img):
         """
         Args:
             img (PIL Image): Image to be flipped.
+            proba (float): probability of image is flipped. Default value is 0.5
```
Comment (fmassa) [inline torchvision/transforms/transforms.py]: the probability is not an argument to the `__call__` method, right?

File: torchvision/transforms/transforms.py [hunk 2]
```diff
@@ -321,7 +321,14 @@ def __repr__(self):
 
 
 class RandomHorizontalFlip(object):
-    """Horizontally flip the given PIL Image randomly with a probability of 0.5."""
+    """Horizontally flip the given PIL Image randomly with a probability.
```
Comment (fmassa) [inline torchvision/transforms/transforms.py]: Maybe you could say something like
```
Horizontally flip the given PIL Image randomly with a given probability
```
?
Same thing for the Vertical flip, what do you think?

File: torchvision/transforms/transforms.py [hunk 3]
```diff
@@ -321,7 +321,14 @@ def __repr__(self):
...
+
+    Args:
+        p (float): probability of image is flipped. Default value is 0.5
```
Comment (fmassa) [inline torchvision/transforms/transforms.py]: maybe we could say instead `probability of the image being flipped. Default value is 0.5`

File: torchvision/transforms/transforms.py (hunk 1 above)
Comment (vfdev-5) [reply torchvision/transforms/transforms.py]: Yes, sorry, forgot to remove this

File: torchvision/transforms/transforms.py (hunk 2 above)
Comment (vfdev-5) [reply torchvision/transforms/transforms.py]: I agree

**Comment by vfdev-5** [comment]
@fmassa I fixed the documentation as you suggested.
What do you think, also about to factorize the code:
```
class RandomSomething(object):
    """
    Args:
        p (float): probability of the image being flipped. Default value is 0.5
    """
    def __init__(self, p=0.5):
        self.p = p
```
as it is present in RandomHorizontalFlip, RandomVerticalFlip and RandomGrayscale ?

**Comment by fmassa** [comment]
@vfdev-5 we could have a common class to handle that, we would lose the docstrings in this case (which are all slightly different).
I think we can leave it like this for the moment.

**Pull request merged**